### PR TITLE
Switch to using fixedLoadLabelAddressIntoReg when loading snippet addresses

### PR DESCRIPTION
**Pull request by aviansie-ben**
The Power codegen contains a number of snippets where the address of an area within the snippet would need to be loaded from within mainline code. Previously, this was achieved by having each snippet manually patch mainline code and/or the pTOC to correspond to the correct address. This had several downsides, including introducing a lot of code duplication and making it much more difficult to exploit POWER10's new PC-relative `paddi` instruction to load such addresses.

To address this, these snippets have been updated to make use of the more generic `fixedLoadLabelAddressIntoReg` helper to load an arbitrary label address into a register. This will work correctly once eclipse/omr#5414 is merged, which moves relocation of pTOC entries containing labels to after snippet emission.

### CODE REVIEW

**Comment by aviansie-ben** [comment]
The interface call snippet changes have been removed, as a race condition was discovered involving them that will need longer to work out. For now, the field watch snippet changes can still be merged.

@zl-wang @gita-omr This is now ready for review.

**Comment by gita-omr** [comment]
@AlenBadel could you please take a look?

**Comment by AlenBadel** [comment]
Very useful simplification using `fixedLoadLabelAddressIntoReg`. LGTM.

**Comment by gita-omr** [comment]
Jenkins test sanity aix,plinux jdk8,jdk11

File: runtime/compiler/p/codegen/J9PPCWatchedInstanceFieldSnippet.cpp [hunk 1]
```diff
@@ -1,65 +0,0 @@
...
-
-#include "codegen/J9PPCWatchedInstanceFieldSnippet.hpp"
-#include "p/codegen/PPCTableOfConstants.hpp" //PTOC
-
-
-uint8_t *TR::J9PPCWatchedInstanceFieldSnippet::emitSnippetBody()
-   {
-   // Call Base class's emitSnippetBody() to populate the snippet
-   // Here Binary Buffer will be ahead by the size of the struct J9JITWatchedInstanceFieldData
-   uint8_t *snippetLocation = (TR::J9WatchedInstanceFieldSnippet::emitSnippetBody() - TR::J9WatchedInstanceFieldSnippet::getLength(0));
```
Comment (gita-omr) [inline runtime/compiler/p/codegen/J9PPCWatchedInstanceFieldSnippet.cpp:31]: I suppose `TR::J9WatchedInstanceFieldSnippet::emitSnippetBody() ` is still going to happen?

Comment (aviansie-ben) [reply runtime/compiler/p/codegen/J9PPCWatchedInstanceFieldSnippet.cpp:31]: Yes. By removing the overridden version of `emitSnippetBody()` from this class, the superclass implementation from `TR::J9WatchedInstanceFieldSnippet` will be called directly.

**Comment by gita-omr** [comment]
Jenkins test sanity aix,plinux jdk8,jdk11

**Pull request merged**